alibi_detect/utils/tensorflow/distance.py

```diff
@@ -83,7 +109,52 @@ def batch_compute_kernel_matrix(
     return k_mat
 
 
-def mmd2_from_kernel_matrix(kernel_mat: tf.Tensor, m: int, permute: bool = False,
+def linear_mmd2(x: tf.Tensor,
+                y: tf.Tensor,
+                kernel: Callable,
+                permute: bool = False) -> Tuple[tf.Tensor, tf.Tensor]:
+    """
+    Compute maximum mean discrepancy (MMD^2) between 2 samples x and y with the
+    linear-time estimator.
+
+    Parameters
+    ----------
+    x
+        Batch of instances of shape [Nx, features].
+    y
+        Batch of instances of shape [Ny, features].
+    kernel
+        Kernel function.
+    permute
+        Whether to permute the row indices. Used for permutation tests.
+    """
+    n = np.shape(x)[0]
+    m = np.shape(y)[0]
+    if n != m:
+        raise RuntimeError("Linear-time estimator requires equal size samples")
+    if not permute:
+        k_xx = kernel(x=x[0::2, :], y=x[1::2, :], diag=True)
```

Comment:
Would this not fail for an odd value of n (and m)? `[0::2, :]`-based subsets would contain extra elements to `[1::2, :]`-based subsets.